src/main/java/com/rabbitmq/client/impl/nio/SslEngineFrameBuilder.java

```diff
@@ -59,18 +66,18 @@ protected boolean somethingToRead() throws IOException {
                         throw new SSLException("buffer overflow in read");
                     case BUFFER_UNDERFLOW:
                         cipherBuffer.compact();
-                        int read = NioHelper.read(channel, cipherBuffer);
-                        if (read == 0) {
-                            return false;
-                        }
-                        cipherBuffer.flip();
-                        break;
+                        underflowHandling = true;
```

Comment:
OK, removing some reading logic and putting it in the class that was already doing, nice.